lib/codeowners/cleaner.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'git'
+require 'logger'
+
+require_relative 'checker/code_owners'
+require_relative 'checker/file_as_array'
+require_relative 'checker/whitelist'
+
+module Codeowners
+  # Cleans a codeowner file by sorting, grouping and removing
+  # useless patterns, all automatically
+  class Cleaner
+    # Get repo metadata and compare with the owners
+    def initialize(repo)
+      @git = Git.open(repo, log: Logger.new(IO::NULL))
+      @repo_dir = repo
```

Comment:
Maybe we can simply instance the @filename directly here and remove the repo reference?
```suggestion
      @file = Checker::CodeOwners.filename(repo)
```

Reply:
Done that! Thanks